src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java

```diff
@@ -482,6 +482,68 @@ void nothingWillBeUpdated() {
                     assertThat(custody.getBookingNumber()).isNotEqualTo("44463B");
                 }
             }
+            @Nested
+            class WhenBookingNumberAlreadySet {
+                @BeforeEach
+                void setup() {
+                    final var custody = aCustodyEvent().getDisposal().getCustody().toBuilder().prisonerNumber("44463B").build();
+                    final var disposal = aCustodyEvent().getDisposal().toBuilder().custody(custody).build();
+                    when(convictionService.getSingleActiveConvictionIdByOffenderIdAndCloseToSentenceDate(anyLong(), any()))
+                            .thenReturn(Result.of(Optional.of(aCustodyEvent().toBuilder().disposal(disposal).build())));
+                }
+
+                @Test
+                void nothingWillBeUpdated() {
+                    var custody = custodyService.updateCustodyBookingNumber("G9542VP", updateCustodyBookingNumber);
+
+                    verify(offenderPrisonerService, never()).refreshOffenderPrisonersFor(any());
```

Comment:
Not sure if you want to verify these are called in the cases where number is new or overwritten?

Reply:
They sort of are above, but I think the test isn't clear to show that. Will change in next PR